### PR TITLE
TSK-953 Refactor adapter for Outbox usage

### PR DESCRIPTION
**Pull request by BerndBreier**

### CODE REVIEW

File: taskana-adapter-camunda-system-connector/src/test/java/pro/taskana/camunda/camundasystemconnector/acceptance/RetrieveCamundaTaskAccTest.java [hunk 1]
```diff
@@ -143,50 +140,49 @@ public void testGetFinishedCamundaTasks() throws ParseException {
         expectedTask.setCreated(formatter.format(date));
         expectedTask.setPriority("50");
         expectedTask.setSuspended("false");
-        
-//        ReferencedTask[] expectedResultBody = new ReferencedTask[] {expectedTask};
-//        ResponseEntity<ReferencedTask[]> expectedResult = new ResponseEntity<ReferencedTask[]>(expectedResultBody, HttpStatus.OK);
-        
+
+        // ReferencedTask[] expectedResultBody = new ReferencedTask[] {expectedTask};
```
Comment (holgerhagen) [inline taskana-adapter-camunda-system-connector/src/test/java/pro/taskana/camunda/camundasystemconnector/acceptance/RetrieveCamundaTaskAccTest.java]: Is this still required?

File: taskana-adapter-taskana-connector/src/main/java/pro/taskana/adapter/taskanaconnector/config/TaskanaSystemConnectorConfiguration.java [hunk 2]
```diff
@@ -79,9 +77,9 @@ public SpringTaskanaEngineConfiguration taskanaEngineConfiguration(
         return new SpringTaskanaEngineConfiguration(taskanaDataSource, true, false, taskanaSchemaName);
     }
 
-    @Bean(name = "taskanaTransactionManager")
-    public PlatformTransactionManager txManager(@Qualifier("taskanaDataSource") DataSource dataSource) {
-        return new DataSourceTransactionManager(dataSource);
-    }
+    // @Bean(name = "taskanaTransactionManager")
```
Comment (holgerhagen) [inline taskana-adapter-taskana-connector/src/main/java/pro/taskana/adapter/taskanaconnector/config/TaskanaSystemConnectorConfiguration.java]: Do we need this?

File: taskana-adapter/src/main/java/pro/taskana/adapter/impl/TaskanaTaskStarter.java [hunk 3]
```diff
@@ -75,139 +65,63 @@ public void retrieveNewReferencedTasksAndCreateCorrespondingTaskanaTasks() {
...
-
-                LOGGER.debug(
-                    "----------retrieveNewReferencedTasksAndCreateCorrespondingTaskanaTasks finished----------------------------");
-                adapterManager.returnConnection(sqlSessionManager);
             }
         }
     }
 
-    @Transactional(rollbackFor = Exception.class)
+    // @Transactional(rollbackFor = Exception.class)
```
Comment (holgerhagen) [inline taskana-adapter/src/main/java/pro/taskana/adapter/impl/TaskanaTaskStarter.java]: ?

File: taskana-adapter-camunda-system-connector/src/test/java/pro/taskana/camunda/camundasystemconnector/acceptance/RetrieveCamundaTaskAccTest.java (hunk 1 above)
Comment (BerndBreier) [reply taskana-adapter-camunda-system-connector/src/test/java/pro/taskana/camunda/camundasystemconnector/acceptance/RetrieveCamundaTaskAccTest.java]: Removed..

File: taskana-adapter/src/main/java/pro/taskana/adapter/impl/TaskanaTaskStarter.java (hunk 3 above)
Comment (BerndBreier) [reply taskana-adapter/src/main/java/pro/taskana/adapter/impl/TaskanaTaskStarter.java]: Removed

**Pull request merged**